src/dbnode/encoding/series_iterator_accumulator.go

```diff
@@ -0,0 +1,281 @@
+	opts SeriesAccumulatorOptions,
+) (SeriesIteratorAccumulator, error) {
+	it := &seriesIteratorAccumulator{
+		id:              iter.ID(),
+		nsID:            iter.Namespace(),
+		seriesIterators: make([]SeriesIterator, 0, 2),
+	}
+
+	if opts.RetainTags {
+		it.tagIterator = iter.Tags().Duplicate()
+	}
+
+	err := it.Add(iter)
+	if err != nil {
+		return nil, err
+	}
+
+	return it, nil
+}
+
+func (it *seriesIteratorAccumulator) Add(iter SeriesIterator) error {
+	if newNs := iter.Namespace(); !newNs.Equal(it.nsID) {
+		return fmt.Errorf("cannot add iterator with namespace %s to accumulator %s",
+			newNs.String(), it.nsID.String())
+	}
+
+	if !iter.Next() || !it.iters.push(iter) {
+		iter.Close()
+		return iter.Err()
+	}
+
+	iterStart := iter.Start()
+	if start := it.start; start.IsZero() || iterStart.Before(start) {
+		it.start = iterStart
+	}
+
+	iterEnd := iter.End()
+	if end := it.end; end.IsZero() || iterEnd.After(end) {
+		it.end = iterEnd
+	}
+
+	it.seriesIterators = append(it.seriesIterators, iter)
+	return nil
+}
+
+func (it *seriesIteratorAccumulator) ID() ident.ID {
+	return it.id
+}
+
+func (it *seriesIteratorAccumulator) Namespace() ident.ID {
+	return it.nsID
+}
+
+func (it *seriesIteratorAccumulator) Tags() ident.TagIterator {
+	if iter := it.tagIterator; iter != nil {
+		return iter
+	}
+	if len(it.seriesIterators) == 0 {
+		return ident.EmptyTagIterator
+	}
+	// NB: the tags for each iterator must be the same, so it's valid to return
+	// from whichever iterator is available.
+	return it.seriesIterators[0].Tags()
+}
+
+func (it *seriesIteratorAccumulator) Start() time.Time {
+	return it.start
+}
+
+func (it *seriesIteratorAccumulator) End() time.Time {
+	return it.end
+}
+
+func (it *seriesIteratorAccumulator) Next() bool {
+	if !it.firstNext {
+		if !it.hasNext() {
+			return false
+		}
+
+		it.moveToNext()
+	}
+
+	it.firstNext = false
+	return it.hasNext()
+}
+
+func (it *seriesIteratorAccumulator) Current() (ts.Datapoint, xtime.Unit, ts.Annotation) {
+	return it.iters.current()
+}
+
+func (it *seriesIteratorAccumulator) Err() error {
+	if it.err != nil {
+		return it.err
+	}
+
+	for _, iter := range it.seriesIterators {
+		if err := iter.Err(); err != nil {
+			it.err = err
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (it *seriesIteratorAccumulator) Close() {
+	if it.isClosed() {
+		return
+	}
+	it.closed = true
+	if it.id != nil {
+		it.id.Finalize()
+		it.id = nil
+	}
+	if it.nsID != nil {
+		it.nsID.Finalize()
+		it.nsID = nil
+	}
+	if it.tagIterator != nil {
+		it.tagIterator.Close()
+		it.tagIterator = nil
+	}
+	it.iters.reset()
+}
+
+func (it *seriesIteratorAccumulator) Replicas() ([]MultiReaderIterator, error) {
+	if l := len(it.seriesIterators); l != 1 {
+		return nil, fmt.Errorf("cannot get replicas for accumulated series "+
+			"iterators: need 1 iterator, have %d", l)
+	}
+
+	return it.seriesIterators[0].Replicas()
+}
+
+func (it *seriesIteratorAccumulator) Reset(opts SeriesIteratorOptions) {
```

Comment:
After complications talked about seems like maybe just it.err = not supported is best route for now.

Reply:
Sounds good; can look into putting this back later if the need arises 👍